src/Symfony/Component/Notifier/Bridge/LinkedIn/LinkedInTransport.php

```diff
@@ -0,0 +1,98 @@
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Smaïne Milianni <smaine.milianni@gmail.com>
+ *
+ * @experimental in 5.2
+ *
+ * @see https://docs.microsoft.com/en-us/linkedin/marketing/integrations/community-management/shares/ugc-post-api#sharecontent
+ */
+final class LinkedInTransport extends AbstractTransport
+{
+    protected const PROTOCOL_VERSION = '2.0.0';
+    protected const PROTOCOL_HEADER = 'X-Restli-Protocol-Version';
+    protected const HOST = 'api.linkedin.com';
+
+    private $authToken;
+    private $accountId;
+
+    public function __construct(string $authToken, string $accountId, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->authToken = $authToken;
+        $this->accountId = $accountId;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('linkedin://%s', $this->getEndpoint());
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage && (null === $message->getOptions() || $message->getOptions() instanceof LinkedInOptions);
+    }
+
+    /**
+     * @see https://docs.microsoft.com/en-us/linkedin/marketing/integrations/community-management/shares/ugc-post-api
+     */
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" (instance of "%s" given).', __CLASS__, ChatMessage::class, \get_class($message)));
+        }
+        if ($message->getOptions() && !$message->getOptions() instanceof LinkedInOptions) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" for options.', __CLASS__, LinkedInOptions::class));
+        }
+
+        if (!($opts = $message->getOptions()) && $notification = $message->getNotification()) {
+            $opts = LinkedInOptions::fromNotification($notification);
+            $opts->author(new AuthorShare($this->accountId));
+        } else {
+            $options['author'] = (new AuthorShare($this->accountId))->author();
+            $options['shareCommentary'] = $message->getSubject();
+        }
+
+        $endpoint = sprintf('https://%s/v2/ugcPosts', $this->getEndpoint());
+
+        $response = $this->client->request('POST', $endpoint, [
+            'auth_bearer' => $this->authToken,
+            'headers' => [self::PROTOCOL_HEADER => self::PROTOCOL_VERSION],
+            'json' => array_filter($opts ? $opts->toArray() : $options),
+        ]);
+
+        if (201 !== $response->getStatusCode()) {
```

Comment:
For common HTTP status code I prefer keep number.
>without to import : Symfony\Component\HttpFoundation\Response;

You should import `Symfony\Component\HttpFoundation\Response;` in the current namespace otherwise it will not work at all

Reply:
Using integer for HTTP status code is what we're doing in core, this should be kept as is.